docs/root/start/sandboxes/locality_load_balancing.rst

```diff
@@ -0,0 +1,141 @@
+
+In this scenario, each locality has 1 healthy replica running. In this case, all the requests should be sent to the locality with the highest priority, which is ``local-1``.
+
+.. code:: shell
+
+    # all requests to local-1
+    $ docker-compose exec -T client-envoy python3 client.py http://localhost:3000/ 100
+    Hello from backend-local-1!: 100, 100.0%
+    Failed: 0
+
+If locality ``local-1`` becomes unhealthy, the requests should be load balanced among the subsequent priority localities, which are ``local-2`` and ``remote-1``. They both have priority 1. We then send 100 requests to the backend cluster, and check the responders.
+
+.. code:: shell
+
+    # bring down local-1
+    $ docker-compose exec -T client-envoy curl -s locality-load-balancing_backend-local-1_1:8000/unhealthy
+    [backend-local-1] Set to unhealthy
+
+    # local-2 and remote-1 localities split the traffic 50:50
+    $ docker-compose exec -T client-envoy python3 client.py http://localhost:3000/ 100
+    Hello from backend-remote-1!: 51, 51.0%
+    Hello from backend-local-2!: 49, 49.0%
+    Failed: 0
+
+Now if ``local-2`` becomes unhealthy also, priority 1 locality is only 50% healthy. Requests will be sent to the ``remote-1`` and ``remote-2``.
+
+.. code:: shell
+
+    # bring down local-2
+    $ docker-compose exec -T client-envoy curl -s locality-load-balancing_backend-local-2_1:8000/unhealthy
+
+    # remote-1 locality receive 100% of the traffic
+    $ docker-compose exec -T client-envoy python3 client.py http://localhost:3000/ 100
+    Hello from backend-remote-1!: actual weight 69.0%
+    Hello from backend-remote-2!: actual weight 31.0%
+    Failed: 0
+
+    # recover local-1 and local-2 after the demo
+    $ docker-compose exec -T client-envoy curl -s locality-load-balancing_backend-local-1_1:8000/healthy
+    $ docker-compose exec -T client-envoy curl -s locality-load-balancing_backend-local-2_1:8000/healthy
+
+
+Step 3: Scenario with multiple replica in the highest priority locality
+***********************************************************************
+
+Continue from previous step. We first scale up the ``local-1`` locality to 5 replicas.
+
+.. code:: shell
+
+    $ docker-compose up --scale backend-local-1=5 -d
+
+We are going to show the scenario that ``local-1`` is just partially healthy. So let's bring down 4 of the replicas in ``local-1``.
+
+.. code:: shell
+
+    # bring down local-1 replicas
+    $ docker-compose exec -T client-envoy curl -s locality-load-balancing_backend-local-1_2:8000/unhealthy
+    $ docker-compose exec -T client-envoy curl -s locality-load-balancing_backend-local-1_3:8000/unhealthy
+    $ docker-compose exec -T client-envoy curl -s locality-load-balancing_backend-local-1_4:8000/unhealthy
+    $ docker-compose exec -T client-envoy curl -s locality-load-balancing_backend-local-1_5:8000/unhealthy
+
+    # check healthiness
+    $ docker-compose exec -T client-envoy curl -s localhost:8001/clusters | grep health_flags
```

Comment:
i think we can use curl directly for this - ie no need to do it inside the container - no ?

Reply:
I prefer doing it inside the client container, to keep the consistent with other commands

Reply:
i guess i suggested this because it means we can remove the curl dependency in the container - ie simplify it

it was also slightly related to my other suggestion of separating the client (ie python script) from the proxy (ie Envoy)

Reply:
One problem is, we have multiple server containers running and they all expose 8000 ports.  If we want to send requests from the hosting machine, they will be all on localhost, and we need to assign them different ports, probably statically. This is not as elegant as sending from the proxy container, where destination can be distinguished with server name.

WDYT?

Reply:
ok - i flagged this one specifically as its the envoy proxy not the backends - i get the point about not having to dynamically assign loads of ports

i guess the thing i care more about is splitting the client from the proxy

conceptually i need to re-remember how this works currently and i find it a bit confusing

tbh - i think my ideal would be to reimplement the python script (the one that batches calls) to curl/bash and then rename client to proxy

i think that its the proxy that is the focus of the sandbox not the client, so best name it clearly

also having the client/proxy mixed up confuses what is going on the backend/s with their un/healthy mecanism - i dont think the client is need at all other than a `client.sh` script on the host